flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/TestFlinkAnonymousTable.java

```diff
@@ -52,12 +53,11 @@ public void testWriteAnonymousTable() throws IOException {
             .option("warehouse", warehouseDir.getAbsolutePath())
             .build();
 
-    Exception exception =
-        Assert.assertThrows(
-            ValidationException.class, () -> table.insertInto(descriptor).execute());
-    Assert.assertNotEquals(
-        "This ObjectIdentifier instance refers to an anonymous object, "
-            + "hence it cannot be converted to ObjectPath and cannot be serialized.",
-        exception.getCause().getMessage());
+    Assertions.assertThatThrownBy(() -> table.insertInto(descriptor).execute())
+        .isInstanceOf(ValidationException.class)
+        .hasCause(
```

Comment:
for cause validation you can do `.isInstanceOf(ValidationException.class).cause().isInstanceOf(TableException.class).hasMessage(..)`